src/main/java/net/snowflake/ingest/streaming/internal/ParquetRowBuffer.java

```diff
@@ -0,0 +1,661 @@
+        type = PrimitiveType.PrimitiveTypeName.INT32;
+        break;
+      case SB8:
+        type = PrimitiveType.PrimitiveTypeName.INT64;
+        break;
+      case SB16:
+        type = PrimitiveType.PrimitiveTypeName.FIXED_LEN_BYTE_ARRAY;
+        break;
+      default:
+        throw new UnsupportedOperationException("Time physical type: " + physicalType);
+    }
+    return type;
+  }
+
+  /**
+   * Get the parquet time unit for column of a Snowflake time logical type.
+   *
+   * @param scale Snowflake column scale
+   * @return column parquet time unit
+   */
+  private static LogicalTypeAnnotation.TimeUnit getTimeUnitFromScale(int scale) {
+    LogicalTypeAnnotation.TimeUnit timeUnit;
+    if (scale <= 3) {
+      timeUnit = LogicalTypeAnnotation.TimeUnit.MILLIS;
+    } else if (scale <= 6) {
+      timeUnit = LogicalTypeAnnotation.TimeUnit.MICROS;
+    } else {
+      timeUnit = LogicalTypeAnnotation.TimeUnit.NANOS;
+    }
+    return timeUnit;
+  }
+
+  @Override
+  boolean hasColumn(String name) {
+    return fieldIndex.containsKey(name);
+  }
+
+  @Override
+  float addRow(
+      Map<String, Object> row,
+      int curRowIndex,
+      Map<String, RowBufferStats> statsMap,
+      Set<String> formattedInputColumnNames) {
+    return addRow(row, data, statsMap, formattedInputColumnNames);
+  }
+
+  @Override
+  float addTempRow(
+      Map<String, Object> row,
+      int curRowIndex,
+      Map<String, RowBufferStats> statsMap,
+      Set<String> formattedInputColumnNames) {
+    return addRow(row, tempData, statsMap, formattedInputColumnNames);
+  }
+
+  /**
+   * Adds a row to the parquet buffer.
+   *
+   * @param row row to add
+   * @param out internal buffer to add to
+   * @param statsMap column stats map
+   * @param inputColumnNames list of input column names after formatting
+   * @return row size
+   */
+  private float addRow(
+      Map<String, Object> row,
+      List<List<Object>> out,
+      Map<String, RowBufferStats> statsMap,
+      Set<String> inputColumnNames) {
+    Object[] indexedRow = new Object[fieldIndex.size()];
+    float size = 0F;
+    for (Map.Entry<String, Object> entry : row.entrySet()) {
+      String key = entry.getKey();
+      Object value = entry.getValue();
+      String columnName = formatColumnName(key);
+      int colIndex = fieldIndex.get(columnName).getSecond();
+      RowBufferStats stats = statsMap.get(columnName);
+      Utils.assertNotNull("Arrow column stats", stats);
+      ColumnMetadata column = fieldIndex.get(columnName).getFirst();
+      if (value != null) {
+        ColumnDescriptor columnDescriptor = schema.getColumns().get(colIndex);
+        PrimitiveType.PrimitiveTypeName typeName =
+            columnDescriptor.getPrimitiveType().getPrimitiveTypeName();
+        ColumnLogicalType logicalType = ColumnLogicalType.valueOf(column.getLogicalType());
+        ColumnPhysicalType physicalType = ColumnPhysicalType.valueOf(column.getPhysicalType());
+        switch (typeName) {
```

Comment:
I think we are missing `case FLOAT`.

Reply:
Do we use FLOAT for something? So far I used only DOUBLE for SF REAL in `getColumnParquetType`.